Abhishek Das submission

abhishekdd opened the pull request:
Implemented all three tasks with test cases.
Requesting a review and advise for improvements if required.

abhishekdd committed:
Added GET endpoint for "/quotes"

abhishekdd committed:
Added POST endpoint for "/quotes", along with test cases

abhishekdd committed:
Added endpoint "/reatime" to handle websocket connection, along with test case

tsegismont commented on src/main/java/io/vertx/gsoc2023/qotd/QuoteOfTheDayVerticle.java:
```diff
@@ -62,6 +63,27 @@ private Router setupRouter() {
 
+    router.post("/quotes").handler(ctx -> {
+      ctx.request()
+        .body()
```
It would be better to add a `BodyHandler` before this handler.

In this small application, this works only because the body is fully received at the same time the request is handled.

abhishekdd replied:
Added the BodyHandler. Also added produces and consumes to enforce checking Accept and ContentType headers.

tsegismont commented on src/main/java/io/vertx/gsoc2023/qotd/QuoteOfTheDayVerticle.java:
```diff
@@ -75,4 +97,11 @@ private Future<RowSet<Row>> selectQuotes() {
+  private Future<RowSet<Row>> insertQuote(String author, String text) {
+    return pgPool.withConnection(connection -> 
+      connection
+      .preparedQuery("INSERT INTO quotes (author, text) VALUES ($1, $2) ")
```
Can you update the query so that you get the identifier generated by the database?

Then, can you change the route handler code to return the generated object to the user?

This should simplify your test as well.

abhishekdd replied:
I had already done it as part of the commit to add /realtime endpoint. You may have been looking at the earlier commit.
However, I did miss to update the test cases before. 
Updated the tests now to use the returned response body for assertion instead of making additional get request.
Thanks for this advise.

tsegismont commented on src/main/java/io/vertx/gsoc2023/qotd/QuoteOfTheDayVerticle.java:
```diff
@@ -75,15 +77,35 @@ private Router setupRouter() {
+        .onSuccess(webSocket -> {
+          ctx.vertx()
+            .eventBus()
+            .localConsumer(REALTIME_EVENT_ADDRESS, message -> {
```
Can you update this so that the local consumer is unregistered when the websocket gets closed?

abhishekdd replied:
Updated it now. 
I also noticed in the document "Every WebSocket automatically registers two handler on the event bus, and when any data are received in this handler, it writes them to itself ".

Is it necessary to unregister those as well?

tsegismont replied:
No, if you the `binaryHandlerID` or the `textHandlerID` of the websocket, Vert.x unregister the local consumer automatically when the websocket is closed

tsegismont reviewed (commented):
Thank you @abhishekdd, good job

abhishekdd committed:
1) Added BodyHandler to POST endpoint /quotes
2) Updated POST endpoint /quotes to respond with the newly created quote, also updated the test cases to utilize this response body
3) Updated endpoint /realtime to unregister the evenbus consumer when websocket is closed

abhishekdd commented:
Another change I made, but forgot to mention in the commit message.
In the test cases I have removed assertions for response status, since those are already handled by expect ResponsePredicate of the webclient.

tsegismont reviewed (approved):
Thank you @abhishekdd LGTM
